src/scheduler.c

```diff
@@ -453,14 +466,22 @@ static void addPref(FILE *file)
  * @param waitingTime: The waiting time of the finished process
  * @param TA: The turnaround time of the finished process
  * @param WTA: The weighted turnaround time of the finished process
- * 
+ *
  * This function writes process finish information to a log file in a specific format.
  */
 static void addFinishLog(FILE *file, int currentTime, int processId, char *state, int arrivalTime, int totalRuntime, int waitingTime, int TA, float WTA)
 {
     fprintf(file, "At time %d process %d %s arr %d total %d remain %d wait %d TA %d WTA %.2f\n", currentTime, processId, state, arrivalTime, totalRuntime, 0, waitingTime, TA, WTA);
 }
 
+
+void processDecremented(int signum)
```

Comment:
If you could:
- Add the function comment.
- Make it static as it is only used in this file scope.
- Put it with the other Signal Handlers for the file organization.